packages/compass-query-bar/src/stores/ai-query-reducer.ts

```diff
@@ -246,6 +258,25 @@ export const runAIQuery = (
 
     // Error when the response is empty or there is nothing to map.
     if (!generatedFields || Object.keys(generatedFields).length === 0) {
+      // The query endpoint may return the aggregation property in addition to filter, project, etc..
+      // It happens when the AI model couldn't generate a query and tried to fulfill a task with the aggregation.
+      if (query.aggregation) {
+        localAppRegistry?.emit('generate-aggregation-from-query', {
+          userInput,
+          aggregation: query.aggregation,
+        });
+        const msg =
+          'Query requires stages from aggregation framework therefore an aggregation was generated.';
+        trackAndLogFailed({
```

Comment:
Have we checked with Julia that an `AI Response Failed` event is expected here? Since it's making a usable pipeline it might not be intuitive as a failure.

Reply:
Good point. I will be on vacation next week. Can you chat about this with Julia and either adjust this PR or open the follow-up PR if this will be merged?

Reply:
Sure thing, I'll ping her now. Not a blocker.

Reply:
Got lgtm here: https://mongodb.slack.com/archives/C05AH835WQN/p1692996996219599?thread_ts=1692738027.149129&cid=C05AH835WQN